src/shell_utils.cpp

```diff
@@ -0,0 +1,169 @@
+#include "shell_utils.h"
+#include <sstream>
+#include <iostream>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+// ---- c-tor for struct Job -----
+Job::Job(const std::string& cmd, int processId, bool running)
+    : command(cmd), pid(processId), isRunning(running) {
+}
+
+std::vector<Job> backgroundJobs; // Vector to store background jobs
```

Comment:
Storing attributes like this in the global scope is bad practice. Something like this indicates that you need a class here